src/Microsoft.DotNet.Darc/src/Darc/Operations/AddBuildToChannelOperation.cs

```diff
@@ -255,13 +244,37 @@ private async Task<int> PromoteBuildAsync(Build build, List<Channel> targetChann
                 return Constants.ErrorCode;
             }
 
+            // Construct the templateParameters and queue time variables. 
+            // Publishing v2 uses variables and v3 uses parameters, so just use the same values for both.
+            var promotionPipelineVariables = new Dictionary<string, string>
+            {
+                { "BarBuildId", build.Id.ToString() },
+                { "PublishingInfraVersion", _options.PublishingInfraVersion.ToString() },
+                { "PromoteToChannelIds", string.Join(",", targetChannels.Select(tch => tch.Id)) },
+                { "EnableSigningValidation", _options.DoSigningValidation.ToString() },
+                { "SigningValidationAdditionalParameters", _options.SigningValidationAdditionalParameters },
+                { "EnableNugetValidation", _options.DoNuGetValidation.ToString() },
+                { "EnableSourceLinkValidation", _options.DoSourcelinkValidation.ToString() },
+                { "PublishInstallersAndChecksums", _options.PublishInstallersAndChecksums.ToString() },
+                { "SymbolPublishingAdditionalParameters", _options.SymbolPublishingAdditionalParameters },
+                { "ArtifactsPublishingAdditionalParameters", _options.ArtifactPublishingAdditionalParameters }
+            };
+
+            if (_options.DoSDLValidation)
+            {
+                promotionPipelineVariables.Add("EnableSDLValidation", _options.DoSDLValidation.ToString());
+                promotionPipelineVariables.Add("SDLValidationCustomParams", _options.SDLValidationParams);
+                promotionPipelineVariables.Add("SDLValidationContinueOnError", _options.SDLValidationContinueOnError);
+            }
+
             int azdoBuildId = await azdoClient.StartNewBuildAsync(build.AzureDevOpsAccount,
                 promotionPipelineInformation.project,
                 promotionPipelineInformation.pipelineId,
                 arcadeSDKSourceBranch,
                 arcadeSDKSourceSHA,
-                queueTimeVariables)
-                .ConfigureAwait(false);
+                promotionPipelineVariables,
+                promotionPipelineVariables
```

Comment:
typo?

Reply:
Nope. We need the same set of values for both parameters and queue time variables.

The variables are used in the old publishing YAML, the template parameters are used in the new version. Passing the same set of values for both things is what makes this change backwards compatible with the current promotion pipeline.